### ISSUE TITLE
[CAL-2084] Payment API endpoint needs better error handling

### ISSUE THREAD

**Issue by alishaz-polymath** [calcom/cal.com#9895]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Currently, we return a 404 suggesting a payment with the provided ID doesn't exist, however, it does exist. It is returned fine from within the /bookings endpoint.  The problem seems to be that there's another issue causing the failure but it is incorrectly categorized as a 404.

The example error report is here:

```
{
    "message": "Payment with id: 3185 not found",
    "error": {
        "issues": [
            {
                "code": "invalid_union",
                "unionErrors": [
                    {
                        "issues": [
                            {
                                "code": "invalid_union",
                                "unionErrors": [
                                    {
                                        "issues": [
                                            {
                                                "code": "invalid_type",
                                                "expected": "string",
                                                "received": "object",
                                                "path": [
                                                    "data"
                                                ],
                                                "message": "Expected string, received object"
                                            }
                                        ],
                                        "name": "ZodError"
                                    },
                                    {
                                        "issues": [
                                            {
                                                "code": "invalid_type",
                                                "expected": "number",
                                                "received": "object",
                                                "path": [
                                                    "data"
                                                ],
                                                "message": "Expected number, received object"
                                            }
                                        ],
                                        "name": "ZodError"
                                    },
                                    {
                                        "issues": [
                                            {
                                                "code": "invalid_type",
                                                "expected": "boolean",
                                                "received": "object",
                                                "path": [
                                                    "data"
                                                ],
                                                "message": "Expected boolean, received object"
                                            }
                                        ],
                                        "name": "ZodError"
                                    }
                                ],
                                "path": [
                                    "data"
                                ],
                                "message": "Invalid input"
                            }
                        ],
                        "name": "ZodError"
                    },
                    {
                        "issues": [
                            {
                                "code": "invalid_type",
                                "expected": "array",
                                "received": "object",
                                "path": [
                                    "data"
                                ],
                                "message": "Expected array, received object"
                            }
                        ],
                        "name": "ZodError"
                    },
.........
                ],
                "path": [
                    "data"
                ],
                "message": "Invalid input"
            }
        ],
        "name": "ZodError"
    }
}
```

### Steps to Reproduce

We can use the **Admin API key** and try:

1. Call the GET request on /v1/bookings/370788
This should contain the payment ID of 3188 in response.
2. Call the GET request on /v1/payments/3188
This would contain the aforementioned error. 

The similar pattern can be used for more bookings and payment IDs. From what I could see, pretty much any payment ID is returning this error, but if you put a number way down the line, the response body changes enough to be able to make out that that's the real 404 response body.

<sub>[CAL-2084](https://linear.app/calcom/issue/CAL-2084/payment-api-endpoint-needs-better-error-handling)</sub>

**Comment by alishaz-polymath** [comment]
cc: @zomars @emrysal

**Comment by abhinaisai2002** [comment]
I am working on this issue

**Comment by fabriziospadaro** [comment]
Any update on this? The /payments endpoint is unusable and returns errors on any API request. Both /payments and /payments/[id] are failing."

**Comment by alishaz-polymath** [comment]
> Any update on this? The /payments endpoint is unusable and returns errors on any API request. Both /payments and /payments/[id] are failing."

Should be solved with the linked PR once it is merged and released 🙏
